src/typescript/azure-openapi-validator/rules/utilities/resolveNestedSchema.ts

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+type DataType =
+  | "integer"
+  | "number"
+  | "string"
+  | "boolean"
+  | "null"
+  | "object"
+  | "array"
+
+type SchemaObject = {
+  type?: DataType
+  items?: SchemaObject
+  properties?: Array<SchemaObject>
+  additionalProperties?: SchemaObject | boolean
+  oneOf?: ReadonlyArray<SchemaObject>
+  $ref?: string
+  allOf?: ReadonlyArray<SchemaObject>
+  anyOf?: ReadonlyArray<SchemaObject>
+}
+
+const skipIfUndefined = <T>(f: (v: T) => T): ((v: T | undefined) => T | undefined) => v =>
```

Comment:
Don't quite understand this file. Could you add more comments about what you are going to achieve?

Reply:
done